### ISSUE TITLE
error with multiple monitors / "primary monitor only"

### ISSUE THREAD

**Issue by tremu** [juvlarN/vibranceGUI#36]

**Comment by juvlarN** [comment]
The error message "Changing the resolution failed: DispChangeBadmode" should only occur when you use the option "Change resolution when ingame". 

Do you have a NVIDIA or AMD GPU?

**Comment by tremu** [comment]
Nvidia GPU, "Change resolution when ingame" is unchecked. If it helps, I have 3 monitors, play CSGO on the primary monitor. All 3 are 1920x1080, as well as my ingame resolution.

**Comment by juvlarN** [comment]
Are you sure all of your monitors can display the resolution (i.e. the hz)? 
Like you can have 144 hz on main monitor but the other two dont support 144hz and then it might fail..

**Comment by JiNova** [comment]
Have the same problem as described here. My primary monitor is 144hz, the secondary however does support up to 60hz max.

**Comment by K0bin** [comment]
One could say I'm experiencing that bug too.
![image](https://user-images.githubusercontent.com/1131720/28529214-4604fa6c-7090-11e7-852c-7ef5a81dbe01.png)

**Comment by juvlarN** [comment]
I'll work on a fix

**Comment by juvlarN** [comment]
Can you try the attached version?
I don't have two monitors at hand now but I've tried to fix it nonetheless. 
Lmk if this works... Otherwise I'll need more information on how to reproduce this issue.

[vibranceGUI-2.3.0.4.zip](https://github.com/juvlarN/vibranceGUI/files/1170572/vibranceGUI-2.3.0.4.zip)

Note that the easiest way to stop these error messages to pop up is to kill the entire vibranceGUI process in Task Manager.

**Comment by JiNova** [comment]
I tried the attached version for 3 days now and as far, as I can tell, the problem seems to be resolved!

**Comment by tremu** [comment]
I'm still getting the same error message (lots of them), but the actual problem is solved: upon alt-tabbing, only the primary monitor's vibrancy is adjusted. Thank you!

**Comment by K0bin** [comment]
Yup, didn't fix it.

![image](https://user-images.githubusercontent.com/1131720/29430252-16fa7e4c-8394-11e7-9e7a-f6f10eb2fdb4.png)

**Comment by Akandesh** [comment]
yep, get this often as well. atleast remove the damn messagebox, I get tabbed down from it.

**Comment by juvlarN** [comment]
Try this version and let me know if it works... Thanks
[vibranceGUI-2.3.0.6.zip](https://github.com/juvlarN/vibranceGUI/files/1412093/vibranceGUI-2.3.0.6.zip)

**Comment by Akandesh** [comment]
Alright I'll try. Maybe it would be nicer if you changed it to a traytip aka ballon tooltip since then we won't get tabbed out and the message still goes through.

**Comment by JiPaix** [comment]
@juvlarN @Akandesh I just ran into the very same problem a minute ago (144hz monitor + game runs at 800x600 and a 1080p 60hz secondary monitor) and this version fixes it.

**Comment by axel37** [comment]
I have the same issue. Started happening a few days ago when i installed my second monitor.

**Comment by giuliannosbrugnera** [comment]
Same issue as well, however I am connecting to my pc through TeamViewer when the error happens.

**Comment by juvlarN** [comment]
@axel37 @giuliannosbrugnera I need more information. 
GPU type, monitor setup (which one is the primary monitor and on which one do you play <-- this might be an issue!), did you recently change the nvidia driver (if it worked before)?

**Comment by juv** [comment]
@axel37 this error message has been removed since a few versions already. Has the issue with color changes changed for you in the new version?

**Comment by juv** [comment]
Hi all,

can you please take a look at this new beta version i have built based on this PR: #140 
I tried to fix this issue once and for all ;-) 

You can download the pre-release .exe here:
https://github.com/juv/vibranceGUI/releases/tag/v2.5.0
